core/src/main/scala/kafka/log/Log.scala

```diff
@@ -262,6 +266,7 @@ class Log(@volatile private var _dir: File,
           val producerStateManager: ProducerStateManager,
           logDirFailureChannel: LogDirFailureChannel,
           private val hadCleanShutdown: Boolean = true,
+          @volatile var topicId: Option[Uuid],
           val keepPartitionMetadataFile: Boolean = true) extends Logging with KafkaMetricsGroup {
```

Comment:
One more thing here. Can we make this a non-optional parameter?

Reply:
I'm wondering about the ZK controller case here. Before we had Uuid.ZERO_UUID represent an unset ID, but I thought it was too easy to use thinking it was a valid ID.

Reply:
Oh. I misunderstood. Were you referring to `keepPartitionMetadataFile`